spec/controllers/users/passwords_controller_spec.rb

```diff
@@ -125,6 +126,31 @@
         expect(flash[:notice]).to eq t('devise.passwords.updated_not_active')
       end
     end
+
+    context 'user with active profile submits valid new password' do
+      let(:profile) { create(:profile, :active, :verified) }
+      let(:user) { profile.user }
+
+      it 'redirects to sign in page' do
+        stub_analytics
+        allow(@analytics).to receive(:track_event)
+
+        params = { password: 'password', reset_password_token: 'foo' }
+
+        allow(User).to receive(:reset_password_by_token).with(params).and_return(user)
+        allow(user).to receive(:reset_password_token).and_return('foo')
+        allow(user).to receive(:errors).and_return({})
+        allow(user).to receive(:password=).with('password')
+
+        put :update, password_form: params
+
+        expect(@analytics).to have_received(:track_event).
+          with('Password reset', user_id: user.uuid)
```

Comment:
I see that the spec above has this same expectation -- should we combine the two specs, perhaps? it looks like they are testing the same things, not 100% overlap in actual spec expectations but no reason I can see that they couldn't be one larger test